### PR TITLE
CC Library APIs 0.1

### PR DESCRIPTION
**Pull request by volfied**
Subject to change, but they work with the current initial libraries implementation.

### CODE REVIEW

File: src/lib/libraries.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
```
Comment (iwehrman) [inline src/lib/libraries.js]: 2015

File: src/lib/libraries.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        layerLib = require("./layer");
+
+    /**
+     * Sends a notification to Photoshop with the created element path
+     *
+     * @param {number} documentID Owner document ID
+     * @param {Array.<number>} layerIDs Affected layer IDs
+     * @param {AdobeLibraryElement} element Created element object
+     * @param {string} path Location of the file in local storage cache
+     *
```
Comment (iwehrman) [inline src/lib/libraries.js]: No new-line here for consistency.

File: src/lib/libraries.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    };
+
+    /**
+     * Sends a notification to Photoshop with the chosen element path
+     *
+     * @param {number} requestRef Reference to initial request sent
+     *                            by Photoshop for relinking
+     * @param {AdobeLibraryElement} element Chosen element object
+     * @param {string} path Location of the file in local storage cache
+     * @param {{id: number, license: bool}} stockData License information on Adobe Stock images
```
Comment (iwehrman) [inline src/lib/libraries.js]: `boolean`

File: src/lib/libraries.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+        if (stockData) {
+            eventData.adobeStock = stockData;
+        }
+
+        return new PlayObject(
+            "spacesLibraryElementChosen",
+            {
+                "null": appReference.current,
+                "json": eventData
```
Comment (iwehrman) [inline src/lib/libraries.js]: It's weird that this is an object but in `createElement` it's a string?

File: src/lib/libraries.js [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+                    "in": {
+                        "_path": path
+                    },
+                    "name": name
+                },
+                "externalPreviewParams": {
+                    "in": {
+                        "_path": previewPath
+                    },
+                    "pixelWidth": 248,
```
Comment (iwehrman) [inline src/lib/libraries.js]: These magic pixel values are weird. Obviously we don't have to fix this now, but do you know anything about whether we can get high-res thumbnails for this? Our UI will be scalable in rems, so we might eventually need breakpoints for choosing high or low res thumbnails.

**Comment by iwehrman** [comment]
A few nits, but otherwise looks fine.

File: src/lib/libraries.js (hunk 2 above)
Comment (volfied) [inline src/lib/libraries.js]: New-line is the standard

File: src/lib/libraries.js (hunk 4 above)
Comment (volfied) [inline src/lib/libraries.js]: Good catch, I wasn't using this API yet.

**Comment by volfied** [comment]
Thanks @iwehrman

File: src/lib/libraries.js (hunk 2 above)
Comment (iwehrman) [inline src/lib/libraries.js]: Hmm... It's standard to separate the English-language description from the JSDoc annotations, but I don't know of any other places where we separate `@param` from `@return`.

**Comment by iwehrman** [comment]
We can keep arguing about whitespace elsewhere. Looks good!

**Pull request merged**